gems/sorbet-runtime/lib/types/props/decorator.rb

```diff
@@ -510,7 +510,12 @@ def prop_defined(name, cls, rules={})
         end
       end
 
-      if rules[:ifunset]
+      # See `prop_get` comment above
+      callers_rely_on_nil_comparison_bug = defined?(BSON) &&
+        defined?(BSON::ByteBuffer) &&
+        rules[:type_object] == T.coerce(BSON::ByteBuffer)
```

Comment:
What do you think about making a `T::Types::Simple` here directly, rather than going through `T.coerce`?

Reply:
No preference, happy to switch!

Reply:
Somewhat related question - any thoughts on adding the BSON gem as a dev-time dependency so that we can add a unit test for this? FWIW it's a nightmare to track down fresh, and it's only because that painful memory is so strong that I guessed this fairly quickly.

Reply:
Resolving since I switched this to a more general strategy to support some other weird legacy stuff.